src/actions.js

```diff
@@ -55,10 +55,43 @@ export const destroyWaiter = (name) => ({
   },
 });
 
+export function waiterResponseHandler(
+  name,
+  { waiterId, responseAction, responseData }
+) {
+  return (dispatch, getState) => {
+    const waiterData = getWaiter(getState(), name);
+
+    // waiter was cleared or destroyed
+    if (waiterData.request === null) {
+      return waiterData;
+    }
+
+    // A new request was made
+    // should not listen to this request anymore
+    if (waiterId !== waiterData.id) {
+      return waiterData;
+    }
+
+    // Request was canceled
+    if (waiterData.isCanceled) {
```

Comment:
Going to leave this here for now. If you wanted to "flesh" out something more advanced, happy to see it.

Reply:
Sounds goof

Reply:
Sounds goof